src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+        classpath_rel = relpathify(classpath_abs)
+        cp_entries.extend(classpath_rel)
+
+        ctx.ensure_output_dirs_exist()
+        
+        tgt, = vts.targets
+        with Timer() as timer:
+          # Step 1: Convert classpath to SemanticDB
+          # ---------------------------------------
+          scalac_classpath_path_entries_abs = self.tool_classpath('workaround-metacp-dependency-classpath')
+          scalac_classpath_path_entries = relpathify(scalac_classpath_path_entries_abs)
+          rsc_outline_dir = relpathalone(ctx.rsc_outline_dir)
+          args = [
+            # NB: Without this setting, rsc will be missing some symbols
+            #     from the scala library.
+            '--include-scala-library-synthetics', # TODO generate these once and cache them
+            # NB: We need to add these extra dependencies in order to be able
+            #     to find symbols used by the scalac jars.
+            '--dependency-classpath', os.pathsep.join(scalac_classpath_path_entries),
+            # NB: The directory to dump the semanticdb jars generated by metacp.
+            # TODO: break this out into a separate job so that we can apply
+            #       it once per 3rdparty target.
+            '--out', rsc_outline_dir,
+            os.pathsep.join(cp_entries),
+          ]
+          metacp_wu = self._runtool(
+            'scala.meta.cli.Metacp',
+            'metacp',
+            args,
+            distribution,
+            tgt=tgt,
+            input_files=(scalac_classpath_path_entries + classpath_rel),
+            output_dir=rsc_outline_dir)
+          metacp_stdout = stdout_contents(metacp_wu)
+          metacp_result = json.loads(metacp_stdout)
+          metai_classpath = []
+          
+          def desandboxify_pantsd_loc(path):
+            # TODO come up with a cleaner way to maybe relpath paths.
+            try:
+              path = relpathalone(path)
+            except Exception:
+              pass
+            pattern = 'process-execution[^{}]+/'.format(re.escape(os.path.sep))
+            return re.split(pattern, path)[-1]
+
+          # TODO when these are generated once, we won't need to collect them here.  
+          metai_classpath.append(desandboxify_pantsd_loc(metacp_result["scalaLibrarySynthetics"]))
+          # NB The json is absolute pathed pointing into either the buildroot or
+          #    the temp directory of the hermetic build. This relativizes the keys.
+          status_elements = {
+            desandboxify_pantsd_loc(k): v
+            for k,v in metacp_result["status"].items()
+          }
+
+          for cp_entry in classpath_rel:
+            metai_classpath.append(desandboxify_pantsd_loc(status_elements[cp_entry]))
+          for cp_entry in jvm_lib_jars_abs:
+            metai_classpath.append(desandboxify_pantsd_loc(status_elements[cp_entry]))
+
+          metacp_classpath_entries = metai_classpath
+
+          # Step 1.5: metai Index the semanticdbs
```

Comment:
Looks like this line contains a typo?

Reply:
Hm. I'm not seeing it, could you be more specific? Maybe that metai isn't capitalized? I'm sure I'll 🤦‍♀️ when you point it out.

Reply:
I was surprised that "Index" is capitalized, being in the middle of the sentence and not being a proper noun. That would be a typo in Russian, but I'm not a native English speaker, so I'm not sure about English.

As for tool names, I personally write Metai, Metacp, Metap, Rsc, etc - capitalizing just the first letter (i.e. basically treating these names / abbreviations as proper nouns). However, I don't think there is any standard spelling here (e.g. Olaf frequently writes tool names in lowercase), so until then any spelling makes sense.